ghci-wrapper/src/Language/Haskell/GhciWrapper.hs

```diff
@@ -94,10 +102,12 @@ close repl = do
   when (e /= ExitSuccess) $ do
     throwIO (userError $ "Language.Haskell.GhciWrapper.close: Interpreter exited with an error (" ++ show e ++ ")")
 
-putExpression :: Interpreter -> String -> IO ()
-putExpression Interpreter{hIn = stdin} e = do
+putExpression :: Interpreter -> Bool -> String -> IO ()
+putExpression Interpreter{hIn = stdin} preserveIt e = do
   hPutStrLn stdin e
+  when preserveIt $ hPutStrLn stdin $ "let " ++ itSafe ++ " = it"
   hPutStrLn stdin marker
+  when preserveIt $ hPutStrLn stdin $ "let it = " ++ itSafe
```

Comment:
Is this ever harmful or could we just always restore `it`?  If it's ok to always restore `it`, then I vote for removing the `preserveIt` argument from this function and do it unconditionally.